### DISCUSSION TITLE
Use schemas from registry rather than locally

### DISCUSSION THREAD

**Discussion by iMacTia** [flipp-oss/deimos#148] [Ideas]
Schemas currently need to be downloaded locally and registered with a producer in order to be used for generating Ruby classes. It would be really cool if `deimos` was able to get them directly from the schema registry instead! Is there any reason why that's not possible at the moment?

**Comment by dorner** [comment]
Hey! 

The reason we recommend having the schema locally is because Avro schemas in particular have backwards compatibility guarantees. If you have your own reader schema, and the writer upstream (e.g.) adds a field in a backwards-compatible way, your system does not need to change at all. You'll just ignore the new field as if it doesn't exist. You're basically sacrificing instant updates for safety in the fact that the new fields can't possibly break your code.

Having said that, I'm definitely not against pulling from the registry itself. In fact, you're the second person to suggest doing so in the past month. It looks like AvroTurf supports doing this for readers but not for writers - you can see some details here: https://github.com/dasch/avro_turf/issues/122 As long as this can be configured, I'm fine with adding it in.

**Reply by iMacTia** [reply]
That is a really good point that I hadn't considered actually, we'd need some sort of version-management of schemas and/or a place where we specify which schemas (and corresponding version) you'd like to download and use.
Good to know `avro_turf` already supports that, thanks for pointing me there!
